cognite_toolkit/_cdf_tk/commands/auth.py

```diff
@@ -524,17 +527,28 @@ def check_identity_provider(self, ToolGlobals: CDFToolConfig, cdf_project: str)
 
     def check_count_group_memberships(self, user_group: GroupList) -> None:
         print("Checking CDF group memberships for the current client configured...")
-
-        table = Table(title="CDF Group ids, Names, and Source Ids")
-        table.add_column("Id", justify="left")
-        table.add_column("Name", justify="left")
-        table.add_column("Source Id", justify="left")
-        for group in user_group:
-            name = group.name
-            if group.name == TOOLKIT_SERVICE_PRINCIPAL_GROUP_NAME:
-                name = f"[bold]{group.name}[/]"
-            table.add_row(str(group.id), name, group.source_id)
-        print(table)
+        has_source_id = any(group.source_id for group in user_group)
+        if has_source_id:
+            table = Table(title="CDF Group ids, Names, and Source Ids")
+            table.add_column("Id", justify="left")
+            table.add_column("Name", justify="left")
+            table.add_column("Source Id", justify="left")
```

Comment:
Better to do the if check here. You are duplicating a more code than necessary.

```suggestion
    if has_source_id:
        table.add_column("Source Id", justify="left")
```